javaslang/src/test/java/javaslang/collection/JavaConvertersTest.java

```diff
@@ -0,0 +1,178 @@
+/*     / \____  _    _  ____   ______  / \ ____  __    _______
+ *    /  /    \/ \  / \/    \ /  /\__\/  //    \/  \  //  /\__\   JΛVΛSLΛNG
+ *  _/  /  /\  \  \/  /  /\  \\__\\  \  //  /\  \ /\\/ \ /__\ \   Copyright 2014-2016 Javaslang, http://javaslang.io
+ * /___/\_/  \_/\____/\_/  \_/\__\/__/\__\_/  \_//  \__/\_____/   Licensed under the Apache License, Version 2.0
+ */
+package javaslang.collection;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class JavaConvertersTest {
+
+    public abstract static class ListViewTest {
```

Comment:
The abstract test contains the methods that are common for immutable and mutable versions.
It is parameterized with a List factory, so we can test our test-cases with standard j.u.List impls and Javaslang List views.

It is important to test against j.u. Collections in order to ensure that our tests are correct and our views behave exactly as standard Java (e.g. when concurrent modifications occur, hashCode, equals and toString etc.)

(Now I will add missing tests...)